src/lib/util/is-email.ts

```diff
@@ -0,0 +1,19 @@
+// Email address matcher.
+// eslint-disable-next-line no-useless-escape
+const matcher = /.+\@.+\..+/;
```

Comment:
This might be enough, but the RFC is a lot more complex and suggests a regex that looks something like this (https://www.emailregex.com/)
```
(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])
```

Reply:
jeah, the only use-case we have for now is to determine whether a use signs-in with a "username" (@ not a valid char) or an email. We are not checking if it is a valid email.